src/main/java/com/google/edith/servlets/GroceryListServlet.java

```diff
@@ -0,0 +1,38 @@
+package com.google.edith;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/grocery-list-query")
+public class GroceryListServlet extends HttpServlet {
+  private UserService userService;
+  private DatastoreService datastoreService;
+  private QueryItems queryItems;
+
+  public GroceryListServlet(
+      UserService userService, DatastoreService datastoreService, QueryItems queryItems) {
+    this.userService = userService;
+    this.datastoreService = datastoreService;
+    this.queryItems = queryItems;
```

Comment:
Do we need the QueryItems as a separate params if they are built using the userService and datastoreService? Or does this help in testing?

Reply:
yes this constructor is used for testing purposes only, otherwise the default constructor is used